commands_buy.go

```diff
@@ -58,5 +59,49 @@ func (b buyCmd) ToAuditEntry() string {
 }
 
 func (b buyCmd) Execute() {
-	consoleLog.Warning("Not implemented: BUY")
+	consoleLog.Debug("Starting Buy")
+	if _, accountExists := accountStore[b.userID]; !accountExists {
+		consoleLog.Errorf("No account for %s exists", b.userID)
+		return
```

Comment:
Use `abortTx` , `abortTxOnError` for control flow in a transaction. The implementation of those functions allows any (future) cleanup to exist in one place.